lib/build.gradle

```diff
@@ -1,25 +1,20 @@
-/*
- * This file was generated by the Gradle 'init' task.
- *
- * This generated file contains a sample Java library project to get you started.
- * For more details take a look at the 'Building Java & JVM projects' chapter in the Gradle
- * User Manual available at https://docs.gradle.org/7.5.1/userguide/building_java_projects.html
- * This project uses @Incubating APIs which are subject to change.
- */
+import com.github.jengelman.gradle.plugins.shadow.tasks.ConfigureShadowRelocation
 
 plugins {
-    // Apply the java-library plugin for API and implementation separation.
     id 'java-library'
     id 'maven-publish'
     id 'signing'
-    id "com.github.johnrengelman.shadow" version "7.0.0"
 
+    id "com.github.johnrengelman.shadow" version "7.1.2"
 }
+
+sourceCompatibility = JavaVersion.VERSION_11
+targetCompatibility = JavaVersion.VERSION_11
```

Comment:
This can be set to Java 8 for wider compatibility with some simple code changes.